jetty-websocket/websocket-jetty-server/src/main/java/org/eclipse/jetty/websocket/server/config/JettyWebSocketConfiguration.java

```diff
@@ -48,39 +48,29 @@
     public JettyWebSocketConfiguration()
     {
         addDependencies(WebXmlConfiguration.class, MetaInfConfiguration.class, WebInfConfiguration.class, FragmentConfiguration.class);
+        addDependents("org.eclipse.jetty.osgi.annotations.AnnotationConfiguration", WebAppConfiguration.class.getName());
+        addDependents("org.eclipse.jetty.annotations.AnnotationConfiguration", WebAppConfiguration.class.getName());
 
-        if (isAvailable("org.eclipse.jetty.osgi.annotations.AnnotationConfiguration"))
-            addDependents("org.eclipse.jetty.osgi.annotations.AnnotationConfiguration", WebAppConfiguration.class.getName());
-        else if (isAvailable("org.eclipse.jetty.annotations.AnnotationConfiguration"))
-            addDependents("org.eclipse.jetty.annotations.AnnotationConfiguration", WebAppConfiguration.class.getName());
-        else
-            throw new RuntimeException("Unable to add AnnotationConfiguration dependent (not present in classpath)");
-
-        protectAndExpose(
-            "org.eclipse.jetty.websocket.api.",
-            "org.eclipse.jetty.websocket.server.",
-            "org.eclipse.jetty.websocket.util.server."); // For WebSocketUpgradeFilter
-
-        hide("org.eclipse.jetty.server.internal.",
-            "org.eclipse.jetty.server.config.");
+        protectAndExpose("org.eclipse.jetty.websocket.api.");
+        protectAndExpose("org.eclipse.jetty.websocket.server.");
+        protectAndExpose("org.eclipse.jetty.websocket.util.server."); // For WebSocketUpgradeFilter
+        hide("org.eclipse.jetty.server.internal.");
+        hide("org.eclipse.jetty.server.config.");
     }
 
     @Override
     public boolean isAvailable()
-    {
-        return isAvailable("org.eclipse.jetty.websocket.common.JettyWebSocketFrame");
-    }
-
-    private boolean isAvailable(String classname)
     {
         try
         {
-            return Loader.loadClass(classname) != null;
+            ClassLoader classLoader = JettyWebSocketConfiguration.class.getClassLoader();
+            return classLoader.loadClass("org.eclipse.jetty.websocket.server.JettyWebSocketServerContainer") != null;
```

Comment:
Ditto - maybe just return true here.
Feel free to improve javadoc on isAvailable if you think it wasn't made clear